src/main/java/upgrade/UpgradePresenter.java

```diff
@@ -0,0 +1,30 @@
+package upgrade;
+
+public class UpgradePresenter implements UpgradeOutputBoundary {
+
+    private String currentEmail;
+    private boolean finishSubscribed;
+
+    public UpgradePresenter(String currentEmail) {
+        this.currentEmail = currentEmail;
+        this.finishSubscribed = false;
```

Comment:
may be redundant as the default value of Boolean is false.

Reply:
I think this should be included in the constructor, though the default value is false

Reply:
Shuhao is right about that finishSubscribed is not need in the constructor to initialize it. But for readability, it's good to include finishSubscribed in the constructor. Leaving it like this is fine.

Reply:
Agree  with @ShuHaoLin2002  and @cyltaracai